### ISSUE TITLE
RuntimeException: Issue calculating the type of the scope of MethodCallExprContext

### ISSUE THREAD

**Issue by tamtam22** [javaparser/javasymbolsolver#156]
**ChangePassword.java**
~~~~
public class ChangePassword extends Activity {
    EditText changePassword_text;

    class RequestChangePasswordTask {
        public void postData() {
            List<NameValuePair> nameValuePairs = new ArrayList(2);
            nameValuePairs.add(new BasicNameValuePair("newpassword", ChangePassword.this.changePassword_text.getText().toString()));
       }
    }
}
~~~~

**Parsing code:**
~~~~
try {
	SymbolReference<MethodDeclaration> symbol = JavaParserFacade.get(cts).solve(node);
	if (symbol.isSolved()) {
		String name = symbol.getCorrespondingDeclaration().getQualifiedName();
		System.out.println("SOLVED: " + name.substring(0, name.lastIndexOf(".")));
	}
} catch (Exception e) {
	if(JavaParserFacade.get(cts).solve(node.getName()).isSolved()) {
		System.out.println(JavaParserFacade.get(cts).solve(node.getName()).getCorrespondingDeclaration().getType().toString());
	} else {
		e.printStackTrace();
	}
}
~~~~
**Output:**
~~~~
**ChangePassword.this.changePassword_text.getText()**
java.lang.RuntimeException: Issur calculating the type of the scope of MethodCallExprContext{wrapped=ChangePassword.this.changePassword_text.getText()}
Caused by: UnsolvedSymbolException{context='unknown', name='changePassword_text', typeSolver=null}

**ChangePassword.this.changePassword_text.getText().toString()**
java.lang.RuntimeException: Issur calculating the type of the scope of MethodCallExprContext{wrapped=ChangePassword.this.changePassword_text.getText().toString()}
Caused by: UnsolvedSymbolException{context='unknown', name='changePassword_text', typeSolver=null}
~~~~

I'm not able to resolve the symbol even though the type is clearly visible from the variable declarator.

**Comment by tamtam22** [comment]
**Error on code:**
`SymbolReference<ValueDeclaration> sr = JavaParserFacade.get(combinedTypeSolver).solve(((MethodCallExpr) node).getName());`

**Incompatible types ERROR**
Required: `SymbolReference <com.github.javaparser.symbolsolver.model.declarations.ValueDeclaration>`
Found: `SymbolReference <capture<? extends com.github.javaparser.symbolsolver.model.declarations.ValueDeclaration>>`

How do i get the symbol reference correctly?

**Comment by 3ch0** [comment]
Regarding your first post, i had the same issue on my current project when i tried to solve a MethodCallExpr symbol (using a visitor). The only solution i found to make it worked was to set jreOnly to false when i added the ReflectionTypeSolver. But i am new to JavaParser and JavaSymbolSolver and maybe there is a better solution.

Hope this answer helped you.

**Comment by ftomassetti** [comment]
The error message contains a typo (Issur instead of Issue). Fixed the typo.

**Comment by Meddan** [comment]
Setting jreOnly to false works, however it cannot solve method calls when the name of the scope is the same as the class of the scope:

```
   A a = new A();
   a.b();
```

gives: Exception in thread "main" java.lang.NoClassDefFoundError: Test/a (wrong name: Test/A)

**Comment by xdrop** [comment]
@tamtam22 @3ch0 @Meddan Can anyone confirm whether this is resolved with #193 ?

**Comment by xdrop** [comment]
Closing this as it's probably been resolved by #193
